rules.py

```diff
@@ -941,38 +941,47 @@ def check(self, element):
                 element.sourceline, element.get("objectId")))
 
 class AllCaps(base.BaseRule):
-    """The Name elements in Candidates, Contests and Person elements should not be in all uppercase.
+    """The Name elements in Candidates, Contests and Person elements should not
+    be in all uppercase.
 
-    If the name elements in Candidates, Contests and Person elements are in uppercase, 
-    the list of objectIds of those elements will be returned to the user as a warning."""
+    If the name elements in Candidates, Contests and Person elements are in
+    uppercase, the list of objectIds of those elements will be returned to
+    the user as a warning.
+    """
 
     def elements(self):
-        return ["Candidate", "CandidateContest", "Person"]
+        return ["Candidate", "CandidateContest", "PartyContest", "Person"]
 
     def check(self, element):
         object_id = element.get("objectId")
         if element.tag == "Candidate":
-            ballot_name = element.find("BallotName")
-            if ballot_name.find("Text") is not None:
-                name = ballot_name.find("Text").text
-                if name is not None and name == name.upper():
-                    raise base.ElectionWarning("Line %d. Candidate %s has name in all upper case letters." % (
-                        element.sourceline, object_id))
+            ballot_name_element = element.find("BallotName")
+            if ballot_name_element is not None:
```

Comment:
I'd suggest getting rid of this nesting by doing the opposite checks, and then returning if true.

e.g.

ballot_name_element = element.find("BallotName")
if ballot_name_element is None:
  return
ballot_name = ballot_name_element.find("Text")
if ballot_name is None:
  return

etc.